pkg/logql/log/labels.go

```diff
@@ -639,32 +632,31 @@ func (b *LabelsBuilder) LabelsResult() LabelsResult {
 		}
 	}
 
-	result := NewLabelsResult(b.buf.String(), hash, labels.New(stream...), labels.New(meta...), labels.New(parsed...))
+	result := NewLabelsResult(lbls.String(), hash, labels.New(stream...), labels.New(meta...), labels.New(parsed...))
 	b.resultCache[hash] = result
 
 	return result
 }
 
-func flattenLabels(buf labels.Labels, many ...labels.Labels) labels.Labels {
+func flattenLabels(many ...labels.Labels) labels.Labels {
 	var size int
 	for _, lbls := range many {
-		size += len(lbls)
+		size += lbls.Len()
 	}
 
-	if buf == nil || cap(buf) < size {
-		buf = make(labels.Labels, 0, size)
-	} else {
-		buf = buf[:0]
-	}
+	b := labels.NewScratchBuilder(size)
```

Comment:
This is tricky in Loki and Prometheus is missing a builder for this. In some cases buf is reused which would equal a `ScratchBuilder.Reset()`. However. I just saw that `flattenLabels` is actually used only once. I'm wondering if we should just move this body into `LabelsResult.Labels()`

Reply:
Good idea, done in 922413e.